controllers/control/k8ssandratask_controller.go

```diff
@@ -150,6 +150,7 @@ func (r *K8ssandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	if dcs, err := filterDcs(kc, kTask.Spec.Datacenters); err != nil {
 		return r.reportInvalidSpec(ctx, kTask, err.Error())
 	} else {
+		patch := client.MergeFrom(kTask.DeepCopy())
```

Comment:
This does not fix the flaky tests, but it's a bug, otherwise `Status.Datacenters` will never be updated.